src/lib/comms/sol-socket-dtls-impl-tinydtls.c

```diff
@@ -655,6 +656,98 @@ get_psk_info(struct dtls_context_t *ctx, const session_t *session,
     return r;
 }
 
+static int
+get_ecdsa_key(struct dtls_context_t *ctx, const session_t *session,
+    const dtls_ecdsa_key_t **result)
+{
+    struct sol_socket_dtls *socket = dtls_get_app_data(ctx);
+    struct sol_network_link_addr addr;
+    static unsigned char ecdsa_priv_key[SOL_DTLS_ECDSA_PRIV_KEY_LEN],
```

Comment:
If the two sockets try to handshake at the same time, a race-condition can happen on this variable indeed; otherwise, no problem.

What do you suggest? The only place I see where this could go into to solve that is the `struct sol_socket_dtls`; is it ok to move it there?